var/spack/repos/builtin/packages/r-knitr/package.py

```diff
@@ -26,10 +29,11 @@ class RKnitr(RPackage):
     depends_on('r@3.1.0:', when='@1.15:1.22', type=('build', 'run'))
     depends_on('r@3.2.3:', when='@1.23:', type=('build', 'run'))
     depends_on('r-evaluate@0.10:', type=('build', 'run'))
-    depends_on('r-digest@:1.17', when='@:1.24', type=('build', 'run'))
-    depends_on('r-formatr@:1.14', when='@:1.24', type=('build', 'run'))
     depends_on('r-highr', type=('build', 'run'))
-    depends_on('r-stringr@0.6:', type=('build', 'run'))
     depends_on('r-markdown', type=('build', 'run'))
+    depends_on('r-stringr@0.6:', type=('build', 'run'))
     depends_on('r-yaml@2.1.19:', type=('build', 'run'))
     depends_on('r-xfun', when='@1.23:', type=('build', 'run'))
+    depends_on('r-xfun@0.15:', when='@1.30:', type=('build', 'run'))
+    depends_on('r-digest@:1.17', when='@:1.24', type=('build', 'run'))
+    depends_on('r-formatr@:1.14', when='@:1.24', type=('build', 'run'))
```

Comment:
I think these are actually wrong? digest and formatr are optional in 1.24.

Reply:
They are wrong. I only repositioned those in this PR so they have been incorrect for a while. The version constraints on those packages should actually be in the `when` clause.